pkg/api/types.go

```diff
@@ -2128,3 +2128,35 @@ type RangeAllocation struct {
 	// a single allocated address (the fifth bit on CIDR 10.0.0.0/8 is 10.0.0.4).
 	Data []byte `json:"data"`
 }
+
+// A ThirdPartyObject is a generic representation of a resource, it is used by add-ons and plugins to add new resource
+// types to the API.  It consists of one or more Versions of the api.
+type ThirdPartyObject struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty" description:"standard object metadata"`
+
+	Description string `json:"description,omitempty" description:"The description of this object"`
+
+	Versions []Version `json:"versions,omitempty" description:"The versions for this third party object"`
```

Comment:
Oh, were you planning to use metadata.name of this resource as the kind? We don't want to do that. We want the Kind to be a CamelCase identifier, whereas our names are all lowercase, can include dashes, etc. Also, I could easily imagine creating multiple custom resources, which must have different resource names themselves, but with the same Kinds, registered in different  API groups.